compiler/rustc_attr_parsing/src/context.rs

```diff
@@ -310,6 +322,24 @@ impl<'f, 'sess: 'f, S: Stage> AcceptContext<'f, 'sess, S> {
             },
         })
     }
+
+    pub(crate) fn expect_no_args(&self, args: &ArgParser<'_>) -> Result<(), ErrorGuaranteed> {
+        if let Some(span) = match args {
```

Comment:
I think for `#[cold]` I implemented this exact method. Could you rebase on that branch? Unfortunately it's taking a bit for everything to merge in. https://github.com/rust-lang/rust/pull/142491